src/services/restClient.ts

```diff
@@ -0,0 +1,34 @@
+const defaultHeader = { "Content-type": "application/json" };
+
+const defaultOptions = { method: "GET", headers: defaultHeader };
+
+const _fetch = async (url: string, options: RequestInit = defaultOptions) => {
+  return fetch(url, options);
+};
+
+export const get = async (url: string, data: any) => {
+  try {
+    if(data) {
+      url += setQueryParams(data)
+    }
+    return await _fetch(url, {
+      ...defaultOptions
+    });
+  } catch (e) {
+    console.error(`Failed to fetch from ${url}`, e);
```

Comment:
Should we throw an exception to display an error on the UI?

Reply:
unless we catch that error and allow the consumer of this service to either handle communicating the error to the user or even just swallowing it then any un-caught error will stop execution. Figure at an early stage just having better error reporting for the console is enough for our current needs.